auth/auth.go

```diff
@@ -179,6 +176,44 @@ func (c *Client) Token(ctx context.Context, cfg *config.MountConfig) (*oauth2.To
 	return &oauth2.Token{AccessToken: gcpSAResp.GetAccessToken()}, nil
 }
 
+func (c *Client) extractSAToken(cfg *config.MountConfig, idPool string) (*authenticationv1.TokenRequestStatus, error) {
+	audienceTokens := map[string]authenticationv1.TokenRequestStatus{}
+	if err := json.Unmarshal([]byte(cfg.PodInfo.ServiceAccountTokens), &audienceTokens); err != nil {
+		return nil, err
+	}
+	for k, v := range audienceTokens {
+		if k == idPool { // Only returns the token if the audience is the workload identity. Other tokens cannot be used.
+			return &v, nil
+		}
+	}
+	return nil, fmt.Errorf("unable to obtain token from driver")
```

Comment:
it may be helpful to tell them which 'audience' was missing by saying that a token for `idPool` was not found

Reply:
Error statement modified